haystack_experimental/core/pipeline/breakpoint.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
This file is only been added to update `_create_agent_snapshot` to create the newly extended `AgentSnapshot` which now includes `tool_execution_decisions`.

Unfortunately, this also required adding in the function `_create_pipeline_snapshot_from_tool_invoker` so that it would use the newly updated one.